### PR TITLE
Lint cleanup

### PR DESCRIPTION
**Pull request by willbush**
Based on https://github.com/NixOS/nixpkgs-vet/pull/124, but this time fewer pedantic lint fixes and cleaner git history.

### CODE REVIEW

**Comment by willbush** [comment]
@Ben-PH feel free to review too! (Github won't let me add you as a reviewer)

File: src/main.rs [hunk 1]
```diff
@@ -1,10 +1,13 @@
-// #![warn(clippy::pedantic)]
-// #![allow(clippy::uninlined_format_args)]
-// #![allow(clippy::enum_glob_use)]
-// #![allow(clippy::module_name_repetitions)]
-// #![allow(clippy::doc_markdown)]
+// Temporarily uncomment to view more pedantic or new nursery lints.
+// #![warn(clippy::pedantic, clippy::nursery)]
```
Comment (Ben-PH) [inline src/main.rs:2]: :+1: 

split these into one line per warning, so that they can be selectively commented/uncommented

File: src/nix_file.rs [hunk 2]
```diff
@@ -563,12 +563,7 @@ mod tests {
             (24, 11, Left("testL")),
         ];
         let expected = BTreeMap::from_iter(cases.map(|(line, column, result)| {
-            (
-                Position { line, column },
-                result
-                    .map(ToString::to_string)
-                    .map_right(|node| node.to_string()),
-            )
+            (Position { line, column }, result.map(ToString::to_string))
```
Comment (Ben-PH) [inline src/nix_file.rs:566]: :+1:

**Review by Ben-PH — COMMENTED** [review]
I like it. 

Minor, inconsequential nit, but the map_or_else functions might be able to use `Path::display` (similar to "instead of `|s| s.to_string()`, use `str::to_string`"), but you do you.

**Comment by willbush** [comment]
> Minor, inconsequential nit, but the map_or_else functions might be able to use `Path::display` (similar to "instead of `|s| s.to_string()`, use `str::to_string`"), but you do you.

Not seeing a way to do that without dropping the `./` appended to the beginning of the path.

**Comment by Ben-PH** [comment]
Missed the `./` part. My mistake.

**Review by infinisil — APPROVED** [review]
LGTM, thanks!

**Pull request merged**